splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceDateFormatter.java

```diff
@@ -14,29 +14,88 @@
 
 package com.splicemachine.derby.utils;
 
+import com.splicemachine.db.iapi.reference.SQLState;
+import java.sql.SQLException;
+import static java.lang.String.format;
+
 public class SpliceDateFormatter {
 
     private String format = null;
-    private java.time.format.DateTimeFormatter formatter;
+    private java.time.format.DateTimeFormatter formatter = null;
 
     public SpliceDateFormatter(String format) {
-        this.format = format != null ? format : "yyyy-M-d";
-        
-        if (format != null) {
+
+        // The default external format of "yyyy-MM-dd" may be used
+        // in error messages, but the corresponding internally used format
+        // is "yyyy-M-d".
+        if (format == null) {
+            format = "yyyy-M-d";
+            this.format = "yyyy-MM-dd";
```

Comment:
These two formats should really be constants. That might also get rid of the comment (that can easily get outdated).

Reply:
The class field can be final, but the other one needs to be variable because it's modified.  The comment is useful to explain there is a difference between the splice date format used in SQL and the date format we need to pass to DateTimeFormatter to get the expected behavior.